test/server.ts

```diff
@@ -237,6 +246,99 @@ test('store server started in the background should use store location wanted by
+  // Promise that completes when all server processes have terminated.
+  let completedPromise: Promise<void> = Promise.resolve()
+  const n = !options.n && options.n !== 0 ? 10 : options.n
+  for (let i = 0; i < n; i++) {
+    const item: ServerProcess = [
+      crossSpawn.spawn(
+        pnpmBin,
+        ['server', 'start',
+          '--store', '../store2',
+          // We don't have to specify --port or --path explicitly because exclusive access to server.json
+          // also fixes the possibility of port collision.
+        ]),
+      createDeferred<void>()]
+    serverProcessList.push(item)
+    byline(item[0].stderr).on('data', (line: Buffer) => console.error(`${item[0].pid}: ${line}`))
+    byline(item[0].stdout).on('data', (line: Buffer) => console.log(`${item[0].pid}: ${line}`))
+    item[0].on('exit', (code: number|null, signal: string|null) => {
+      if (options.onProcessClosed) {
+        (options.onProcessClosed)(item[0], code, signal)
+      }
+      for (let j = 0; j < serverProcessList.length; j++) {
+        if (serverProcessList[j][0] === item[0]) {
+          serverProcessList.splice(j, 1)
+          break
+        }
+      }
+      if (!signal && code !== 0) {
+        item[1].reject(new Error(`A server process unexpectedly exited with (code, signal) = (${code}, ${signal})`))
+      } else {
+        item[1].resolve()
+      }
+    });
+    completedPromise = completedPromise.then(() => item[1].promise)
+  }
+  await Promise.all([
+    (async () => {
+      await completedPromise
+      // Don't fire timeout if all server processes completed for some reason.
+      if (timeoutPromise !== null) {
+        timeoutPromise.cancel()
+      }
+    })(),
+    (async () => {
+      try {
+        await timeoutPromise
+      } catch (error) {
+        if (error instanceof CancelError) {
+          // All child processes completed, don't fire timeout or try to kill processes.
+          return
+        }
+        // This should never happen so it's OK if we don't kill remaining server processes in this path.
+        throw error
+      }
+      timeoutPromise = null
+      await Promise.all(serverProcessList.map(async (item) => {
+        // Use SIGINT so that the process can delete server.json for the next test.
+        await kill(item[0].pid, 'SIGINT')
+        await item[1].promise
+      }))
+    })(),
+  ])
+}
+
+test('parallel server starts against the same store should result in only one server process existing after 10 seconds', async (t: tape.Test) => {
+  // This test fails if a server process exits with non-zero code.
+  // It is OK if server processes exit due to some bug (not because they detected another server process won the race)
+  // because other tests should capture these bugs.
+
+  const n = 5
+  // Plan that n - 1 of n server processes will close within 10 seconds.
+  t.plan(n - 1)
```

Comment:
CI fails because the plan should be 5.

4 process exit checks + 1 server.json exists check at line 339

Reply:
Aah that's how it works, I thought it was only counting t.pass